prow/plugins/goose/goose.go

```diff
@@ -0,0 +1,216 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package goose adds goose images to an issue or PR in response to a /honk comment
+package goose
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+var (
+	match = regexp.MustCompile(`(?mi)^/(honk)\s*$`)
+	honk  = &realGaggle{
+		url: "https://api.unsplash.com/photos/random?query=goose",
+	}
+)
+
+const (
+	pluginName = "goose"
+)
+
+func init() {
+	plugins.RegisterGenericCommentHandler(pluginName, handleGenericComment, helpProvider)
+}
+
+func helpProvider(config *plugins.Configuration, enabledRepos []string) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: "The goose plugin adds a goose image to an issue or PR in response to the `/honk` command.",
+		Config: map[string]string{
+			"": fmt.Sprintf("The goose plugin uses an api key for unsplash.com stored in %s.", config.Goose.KeyPath),
+		},
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/honk",
+		Description: "Add a goose image to the issue or PR",
+		Featured:    false,
+		WhoCanUse:   "Anyone",
+		Examples:    []string{"/honk"},
+	})
+	return pluginHelp, nil
+}
+
+type githubClient interface {
+	CreateComment(owner, repo string, number int, comment string) error
+}
+
+type gaggle interface {
+	readGoose() (string, error)
+}
+
+type realGaggle struct {
+	url     string
+	lock    sync.RWMutex
+	update  time.Time
+	key     string
+	keyPath string
+}
+
+func (c *realGaggle) setKey(keyPath string, log *logrus.Entry) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+	if !time.Now().After(c.update) {
+		return
+	}
+	c.update = time.Now().Add(1 * time.Minute)
+	if keyPath == "" {
+		c.key = ""
+		return
+	}
+	b, err := ioutil.ReadFile(keyPath)
+	if err == nil {
+		c.key = strings.TrimSpace(string(b))
+		return
+	}
+	log.WithError(err).Errorf("failed to read key at %s", keyPath)
+	c.key = ""
+}
+
+type gooseResult struct {
+	ID     string   `json:"id"`
+	Images imageSet `json:"urls"`
+}
+
+type imageSet struct {
+	Raw     string `json:"raw"`
+	Full    string `json:"full"`
+	Regular string `json:"regular"`
+	Small   string `json:"small"`
+	Thumb   string `json:"thumb"`
+}
+
+func (gr gooseResult) Format() (string, error) {
+	if gr.Images.Small == "" {
+		return "", errors.New("empty image url")
+	}
+	img, err := url.Parse(gr.Images.Small)
+	if err != nil {
+		return "", fmt.Errorf("invalid image url %s: %v", gr.Images.Small, err)
+	}
+
+	return fmt.Sprintf("\n![goose image](%s)", img), nil
+}
+
+func (g *realGaggle) URL() string {
+	g.lock.RLock()
+	defer g.lock.RUnlock()
+	uri := string(g.url)
+	if g.key != "" {
```

Comment:
Do you have an access key we can put in the cluster? It doesn't look like https://api.unsplash.com/photos/random?query=goose works incognito

Reply:
Okay I have the info I need (and I'm oncall tomorrow), will work on getting this setup.